src/Common/ZooKeeper/ZooKeeperNodeCache.cpp

```diff
@@ -91,7 +91,9 @@ ZooKeeperNodeCache::ZNode ZooKeeperNodeCache::get(const std::string & path, Coor
 
     /// Node doesn't exist. We must set a watch on node creation (because it wasn't set by tryGetWatch).
 
-    result.exists = zookeeper->existsWatch(path, &result.stat, watch_callback);
+    std::string get_path_value;
+    result.exists = zookeeper->tryGetWatch(path, get_path_value, &result.stat, watch_callback);
```

Comment:
This code contradicts the comment above.

Reply:
@alexey-milovidov In this case if the Node doesn't exist, should we directly return the result instead of setting a watch?

Reply:
I'm not sure, need to read the logic in code...